pkg/components/rook/component.go

```diff
@@ -54,35 +109,26 @@ func (c *component) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContex
 }
 
 func (c *component) RenderManifests() (map[string]string, error) {
-	// Generate YAML for namespace.
-	namespaceStr, err := util.RenderTemplate(namespace, c)
+	helmChart, err := util.LoadChartFromAssets("/components/rook-ceph")
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to render template")
-	}
-
-	// Generate YAML for RBAC resources.
-	rbacStr, err := util.RenderTemplate(rbac, c)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to render template")
+		return nil, fmt.Errorf("load chart from assets: %w", err)
 	}
 
 	// Generate YAML for the Rook operator deployment.
 	c.TolerationsRaw, err = util.RenderTolerations(c.Tolerations)
+	c.NodeSelectorString = convertNodeSelector(c.NodeSelectors)
+
+	values, err := util.RenderTemplate(chartValuesTmpl, c)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to marshal operator tolerations")
+		return nil, fmt.Errorf("render chart values template: %w", err)
 	}
 
-	deploymentStr, err := util.RenderTemplate(deploymentOperator, c)
+	renderedFiles, err := util.RenderChart(helmChart, name, c.Metadata().Namespace, values)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to render template")
+		return nil, fmt.Errorf("render chart: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("rendering chart failed: %w", err)
```

Reply:
This seem silly to me. Is there some convention to use present vs continuous terms or skip `failed` keyword?

Reply:
I would ideally want to skip adding the keyword because if we are returning the error it should be assumed something has failed.